app/routes.py

```diff
@@ -1,2 +1,184 @@
+from flask import Blueprint, jsonify, make_response, request
+from app.models.task import Task 
+from app.models.goal import Goal
+from app import db
+from datetime import date
+import requests
+import os
+from dotenv import load_dotenv
 
+load_dotenv()
+
+task_bp = Blueprint("task_bp", __name__, url_prefix="/tasks")
+goal_bp = Blueprint("goal_bp", __name__, url_prefix="/goals")
+
+@task_bp.route("", methods=["GET", "POST"])
+def handle_all_tasks():
+    tasks_response = []
+    
+    #WV2: Get Tasks Asc, Get Tasks Desc
+    if request.method == "GET":
+        if request.args.get('sort') == 'asc':
+            tasks = Task.query.order_by(Task.title.asc()).all()
+        elif request.args.get('sort') == 'desc':
+            tasks = Task.query.order_by(Task.title.desc()).all()
+
+        #WV1: No Saved Tasks, One Saved Tasks
+        else:
+            tasks = Task.query.all()     
+        for task in tasks:
+            tasks_response.append(task.to_dict())
+        return jsonify(tasks_response), 200 
+
+    #WV1: Create Task With None Completed_At, Create Task - Title, Create Task - Description, Create Task - Completed_At
+    #WV3: Create Task Wtih Valid Completed_At
+    if request.method == "POST":
+        request_body = request.get_json()
+        try:
+            new_task = Task(title = request_body["title"], 
+                            description = request_body["description"],
+                            completed_at = request_body["completed_at"])
+        except KeyError:
+            return ({'details': 'Invalid data'}, 400)        
+        db.session.add(new_task)
+        db.session.commit()
+        return jsonify({"task":new_task.to_dict()}), 201
+
+@task_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_single_task(task_id):
+    task_id = int(task_id)
+    task = Task.query.get(task_id)
+
+    #WV1: Get Task, Get Task Not Found
+    if request.method == "GET":
+        if task is None:
+            return make_response("", 404)
+        return jsonify({"task": task.to_dict()}), 200 
+
+    #WV1: Update Task, Update Task Not Found
+    #WV3: Update Task With Valid Completed At
+    elif request.method == "PUT":
+        input_data = request.get_json()
+        if task is None:
+            return make_response("", 404)
+        task.title = input_data["title"]
+        task.description = input_data["description"]
+        db.session.commit()
+        return ({"task":task.to_dict()}), 200
+
+    #WV1: Delete Task, Delete Task Not Found
+    elif request.method == "DELETE":
+        if task is None:
+            return make_response("", 404)
+        else:
+            db.session.delete(task)
+            db.session.commit()
+            return ({'details': f'Task {task_id} "{task.title}" successfully deleted'}), 200
+
+#WV3: Mark Comp on Incom, Mark Incom on Comp, Mark Comp on Comp, Marck Comp on Missing, Mark Incom on Missing 
+@task_bp.route("/<task_id>/<completion_status>", methods=["PATCH"])
```

Comment:
Very cool idea making the completion status a dynamic part of the route! I hadn't considered doing it that way!